src/validation.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  type DscvrFramesRequest,
+  type DscvrValidationResponse,
+  dscvrClientProtocolPrefix,
+  DscvrUntrustedData,
+} from './types';
+import { DEFAULT_DSCVR_API_URL } from './default';
+import { useUrqlClient } from './clients';
+import { validationQuery } from './queries/validation.query';
+
+type ValidationType = string | number | bigint | null | undefined;
+interface ValidatedQueryResult {
+  user: {
+    id: string;
+  };
+  content?: {
+    id: string;
+  };
+  state: string | null;
+  inputText: string | null;
+  timestamp: string;
+  url: string;
+  buttonIndex: number;
+}
```

Comment:
maybe we could move these to `./types` and import from there to make this code more cleaner?

Reply:
Done